LiteCore/Support/Logging.cc

```diff
@@ -656,13 +656,29 @@ namespace litecore {
         va_end(args);
     }
 
+    void Logging::_logv(LogLevel level, const char* format, va_list args) const {
+        _domain.computeLevel();
+
+        // Argument format is checked by __printflike(3, 0); it is guaranteed a literal constant.
+        // Here, we want to add a prefix to the format. This prefix is assigned by
+        // derived classes. It is generated from the LiteCore source, appendKeyValuePrefix,
+        // and we can trust that it does not include format specifier, '%'.
+        // We will pass the prefixedFormat to _domain.vlog by ignoring "-Wformat-nonliteral."
+
+        std::stringstream prefixOutput;
+        addKeyValuePairs(prefixOutput);
+        std::string prefixedFormat = prefixOutput.str();
+        DebugAssert(prefixedFormat.find('%') == std::string::npos);
+
+        if ( prefixedFormat.empty() ) prefixedFormat = format;
+        else
+            (prefixedFormat += " ") += format;
```

Comment:
If you already have a stringstream why do this?  Why not continue to use the stream and then take it at the end?

```
if( prefixOutput.gcount() > 0) prefixOutput << " ";
prefixOutput << format;
```

Reply:
Thanks for reminding of gcount. I was trying to avoid calling str() mehod.
Will do it.

Reply:
Well, gcount() is the count of istream. pcount no longer available for stringstream.

Reply:
??? What does this mean?

Reply:
gcount is inherited from istream. It's the count aa being an input stream, "Returns the number of characters extracted by the last unformatted input operation performed on the object."
std::strstream exposes the pcount, the put count, but strstream is deprecated.

Reply:
So...where does that leave this PR?  Are we going to remove all instances of strstream?

Reply:
Wait...we are talking about stringstream, not strstream

Reply:
Maybe the correct method is `tellp` rather than `gcount`